src/build_function.jl

```diff
@@ -108,7 +108,11 @@ function _build_function(target::JuliaTarget, op, args...;
                          states = LazyState(),
                          linenumbers = true,
                          wrap_code = nothing,
-                         cse = false, kwargs...)
+                         cse = false,
+                         nanmath = false,
```

Comment:
is there a reason to not default to true?

Reply:
Because defaulting to NaNMath would break our implicit support for complex numbers. See https://github.com/JuliaMath/NaNMath.jl/issues/66

Reply:
Can we just check `if eltype(...) <: AbstractComplex ...` in the replaced function? I think it would be pretty important to have this as the default.